lib/Db/LocalAttachmentMapper.php

```diff
@@ -126,6 +127,7 @@ public function deleteForLocalMailbox(int $localMessageId): void {
 			$this->db->rollBack();
 			throw $e;
 		}
+		$this->db->commit();
```

Comment:
This has to be investigated further. Transaction have to be committed manually and there are many instances were it was forgotten.

Reply:
I fixed some instances I found during my testing of the frontend.

Reply:
would it make sense to have a trait or function that does the repetitive pattern for us?

something like a

```php

function doTransactional(callable $fn, IDBConnection $db) {
  $db->beginTransaction();
  try {
    $result = $fn();
    $db->commit();
    return $result;
  } catch (Throwable $e) {
    $db->rollback();
    throw $e;
}
```

in https://github.com/nextcloud/mail/blob/main/lib/functions.php?

Reply:
Great idea!

Reply:
https://github.com/nextcloud/server/pull/31205 I've also submit this upstream as a proposal. We wouldn't be able to use this as new APIs are not backported. As a short term solution we need this construct in our app as well.